src/platforms/hosted/jlink.c

```diff
@@ -295,10 +298,15 @@ static bool jlink_get_version(void)
 	if (version_length > sizeof(jlink.fw_version))
 		return false;
 
-	/* Read vesion string directly into jlink.version */
+	/* Read version string directly into jlink.version */
 	bmda_usb_transfer(bmda_probe_info.usb_link, NULL, 0, jlink.fw_version, version_length, JLINK_USB_TIMEOUT);
 	jlink.fw_version[version_length - 1U] = '\0'; /* Ensure null termination */
 
+	/* Replace first NUL separating platform and copyright string */
+	char *firstnul = strchr(jlink.fw_version, 0);
+	if (firstnul != NULL)
+		*firstnul = '\n';
```

Comment:
I missed the fact the version string contained the copyright a swell the first time, how about that, took me a second to understand what this was doing :sweat_smile:...

There's a subtle edge case here, that I think we should guard for. If a J-Link probe, be it via a firmware bug, or just inconsistent implementation, reports a firmware version that does not have a null as a separator, or termination, this code will replace the null termination we just ensured before it, which could lead to a bad time (from the `strchr` reference: _"The terminating null character is considered to be a part of the string and can be found when searching for '\0'."_ )
We can't just move this code to before the "Ensure null termination" because on this particular edge case, the lack of null termination would be bad news for `strchr` too, so I think our best bet would be to add a check to make sure we don't replace it after

```suggestion
	/* Ensure null termination */
	char *const null_termination = jlink.fw_version + version_length - 1U;
	*null_termination = '\0';

	/* Replace NULL separating version and copyright string, if it exists */
	char *const null_separator = strchr(jlink.fw_version, '\0');
	if (null_separator != NULL && null_separator < null_termination)
		*null_separator = '\n';
```

Reply:
Yes, string content uncertainty also occured to me. If this were C++, I could just ask to print a std::string of the size reported by JLink firmware, but alas. In BMDA logging is hidden behind macros, so it's not like I can call `write()` for fixed bytes and then it dumps everything with literal NULs to stdout.
Your logic looks clean enough, will apply. Maybe I forgot that I wanted to check where that nul byte found by strchr is located, and didn't want to perform pointer comparison, as it may be unspecified.